src/main/java/blusunrize/immersiveengineering/api/crafting/RefineryRecipe.java

```diff
@@ -54,25 +54,25 @@ public static RefineryRecipe findRecipe(FluidStack input0, FluidStack input1)
 	{
 		for(RefineryRecipe recipe : recipeList)
 		{
-			if(input0!=null)
+			if(!input0.isEmpty())
 			{
-				if(recipe.input0!=null&&input0.containsFluid(recipe.input0))
+				if((!recipe.input0.isEmpty())&&input0.containsFluid(recipe.input0))
```

Comment:
We generally don't put brackets around statements just because they're inverted, unless it significantly improves readability or is syntactically necessary (e.g. `!(a>b)`, and yes, there's a use case for that).

Reply:
there is